utils/osgenerator/osgenerator.go

```diff
@@ -36,16 +38,31 @@ func GenerateOS(filename, version string, unit rune, size int) error {
 	if err != nil {
 		return err
 	}
-	data := bytes.NewBuffer([]byte(version + "\n" + time.Now().String() + "\n"))
-	data.Grow(bufferSize + len(data.Bytes()))
 	buf := make([]byte, bufferSize)
 	rand.Read(buf)
-	data.Write(buf)
-	hash := md5.Sum(data.Bytes())
-	data.Write(hash[:])
+	hash := md5.Sum(buf)
+	cookieBuf := make([]byte, 16)
+	rand.Read(cookieBuf)
+	mockOs := &pb.MockOS{
+		Version: version,
+		Cookie:  fmt.Sprintf("%x", cookieBuf),
+		Data:    buf,
+		Hash:    hash[:],
+	}
+	out, err := proto.Marshal(mockOs)
```

Comment:
optional (up to you): I would do it as follow if not reusing the var:

```suggestion
	out, err := proto.Marshal(&pb.MockOS{
		Version: version,
		Cookie:  fmt.Sprintf("%x", cookieBuf),
		Data:    buf,
		Hash:    hash[:],
	})
```